File: ios/MyNativeModule.m
//
//  MyNativeModule.m
//  ChatAppNew
//
//  Created by Sourav Mahanty on 11/12/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MyNativeModule, NSObject)

RCT_EXTERN_METHOD(addNumbers:(nonnull NSNumber)a
                  b:(nonnull NSNumber)b
                  resolver:(RCTPromiseResolveBlock)resolver
                  rejecter:(RCTPromiseRejectBlock)rejecter)

@end

File: ios/MyNativeModule.swift
//
//  MyNativeModule.swift
//  ChatAppNew
//
//  Created by Sourav Mahanty on 11/12/25.
//

import Foundation

@objc(MyNativeModule)
class MyNativeModule: NSObject {

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc
  func addNumbers(_ a: NSNumber, b: NSNumber, resolver: RCTPromiseResolveBlock, rejecter: RCTPromiseRejectBlock) {
    let result = a.intValue + b.intValue
    resolver(result)
  }
}
